ansible/roles/openshift_setup/tasks/main.yml

```diff
@@ -196,7 +197,36 @@
     set_fact:
       generate_config: "{{ use_custom_config and (not master_config_stat.stat.exists or not console_ssl_stat.stat.exists) }}"
 
-  - name: Create command line for oc cluster up execution
+  # Looks like we need to keep the directory oc uses for config files to /var/lib/origin/openshift.local.config
+  # Tried to use /tmp/openshift.local.clusterup as we do for 3.10
+  # Ran into issues in ec2 deployments where we use a custom httpasswd file
+  # oc was unable to read the path to the httpasswd file, assuming something was hardcoded perhaps in oc code, unsure?
+  - name: Hack for older environments
+    set_fact:
+      oc_cluster_up_base_dir: /var/lib/origin/openshift.local.config
+      oc_host_config_dir: /var/lib/origin/openshift.local.config
+      cluster_master_config_file: /var/lib/origin/openshift.local.config/master/master-config.yaml
+      cluster_auth_htpasswd_file: /var/lib/origin/openshift.local.config/master/catasb_auth_htpasswd
+    when:
+    - ("v3.9" in origin_image_tag) or
+        ("v3.7" in origin_image_tag) or
+        ("v3.6" in origin_image_tag)
+
+
+  - name: Create command line for oc cluster up execution for older oc clients, up to 3.9
+    set_fact:
+      oc_cluster_up_cmd: >-
+        {{ oc_cmd }} cluster up
+        --routing-suffix={{ openshift_routing_suffix }}
+        --public-hostname={{ hostname }}
+        --host-pv-dir={{ persistent_vol_mount_point }}
+        --image={{ origin_image_name }}
+        --version={{ origin_image_tag }}
+        {% if use_custom_config %}--host-config-dir={{ oc_host_config_dir }}{% endif %}
+        --service-catalog=true
+
+  # overwrite oc_cluster_up_cmd when using v3.10 or newer
+  - name: Create command line for oc cluster up execution for newer oc clients, 3.10+
```

Comment:
This is good for now. I think we should definitely start breaking out these tasks into individual files for each version so that we don't have to overwrite facts like this.